src/media_type.rs

```diff
@@ -160,6 +189,30 @@ impl MediaType {
     }
   }
 
+  /// Supports file extensions that are made up of multiple extensions,
+  /// eg. "ts" resolves to TypeScript and "d.ts" resolves to Dts.
+  /// `file_extension` must be lower case.
+  pub fn from_file_extension<S: AsRef<str>>(file_extension: S) -> Self {
+    match file_extension.as_ref() {
+      "ts" => Self::TypeScript,
+      "d.ts" => Self::Dts,
```

Comment:
I think this is actually probably not a good idea and we shouldn't have this method. It would be less error prone for someone to provide a dummy filename in this scenario like `file.d.ts` because they would have to remember to get out `.d.ts` instead of `.ts`.

The rest of the changes look good though.

Reply:
Makes sense, removed `from_file_extension()` in favor of `pub on_path()`